tests/unit_tests/test_sender.py

```diff
@@ -23,3 +26,37 @@ def test_config_save_preserve_order(tmp_path, test_settings):
     saved_config.pop("wandb_version")
 
     assert saved_config == original_config
+
+
+@pytest.mark.parametrize(
+    "file_path",
+    [
+        "file_with_*.txt",
+        "file_with_?.txt",
+        "file_with_[.txt",
+    ],
+)
+def test_send_file_with_glob_characters_is_escaped(tmp_path, test_settings, file_path):
+    dir_watcher = MagicMock()
+    settings = test_settings({"x_files_dir": str(tmp_path)})
```

Comment:
i don't think you need to use test_settings, just use regular settings, i will be removing this fixture